src/VisualStudio/Core/Def/Implementation/EditAndContinue/VsENCRebuildableProjectImpl.cs

```diff
@@ -297,6 +299,8 @@ public int StopDebuggingPE()
                 // Avoid ending the debug session if it has already been ended.
                 if (_encService.DebuggingSession != null)
                 {
+                    _encService.OnBeforeDebuggingStateChanged(DebuggingState.Run, DebuggingState.Design);
```

Comment:
What if we finish the debugging session during the break mode? It would be okay if we don't rely on the previous state, though.